src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -475,34 +477,87 @@ internal BigInteger(int n, uint[]? rgu)
 
             int len;
 
-            // Try to conserve space as much as possible by checking for wasted leading uint[] entries
-            // sometimes the uint[] has leading zeros from bit manipulation operations & and ^
+            // Try to conserve space as much as possible by checking for wasted leading uint entries
+            // (sometimes the uint[] has leading zeros from bit manipulation operations & and ^).
             for (len = value.Length; len > 0 && value[len - 1] == 0; len--);
 
             if (len == 0)
+            {
                 this = s_bnZeroInt;
+            }
             // Values like (Int32.MaxValue+1) are stored as "0x80000000" and as such cannot be packed into _sign
             else if (len == 1 && value[0] < kuMaskHighBit)
             {
                 _sign = (negative ? -(int)value[0] : (int)value[0]);
                 _bits = null;
-                // Although Int32.MinValue fits in _sign, we represent this case differently for negate
-                if (_sign == int.MinValue)
+                if (_sign == int.MinValue) // although Int32.MinValue fits in _sign, we represent this case differently for negate
+                {
                     this = s_bnMinInt;
+                }
             }
             else
             {
                 _sign = negative ? -1 : +1;
-                _bits = new uint[len];
-                Array.Copy(value, _bits, len);
+                if (transfersOwnership && value.Length == len)
+                {
+                    _bits = value;
+                }
+                else
+                {
+                    _bits = new uint[len];
+                    Array.Copy(value, _bits, len);
+                }
             }
+
+            AssertValid();
+        }
+
+        /// <summary>
+        /// Constructor used during bit manipulation and arithmetic.
+        /// When possible the span will be packed into _sign to conserve space.
+        /// </summary>
+        /// <param name="value">The absolute value of the number</param>
+        /// <param name="negative">The bool indicating the sign of the value.</param>
+        private BigInteger(Span<uint> value, bool negative)
+        {
+            // Try to conserve space as much as possible by checking for wasted leading uint entries
+            // (sometimes the span has leading zeros from bit manipulation operations & and ^).
+            if (!value.IsEmpty && value[^1] == 0)
+            {
+                int len = value.Length - 1;
+                while (len > 0 && value[len - 1] == 0)
+                {
+                    len--;
+                }
+                value = value.Slice(0, len);
+            }
+
+            if (value.IsEmpty)
+            {
+                this = s_bnZeroInt;
+            }
+            else if (value.Length == 1 && value[0] < kuMaskHighBit) // values like (Int32.MaxValue+1) are stored as "0x80000000" and as such cannot be packed into _sign
```

Comment:
Personally, I'm never a fan of afterthought comments like this.  I particularly don't like that this one adds horizontal scrolling to the file on GitHub's default width... otherwise I wouldn't have said anything since it's mostly tomato/tomato.

Reply:
Booooo. :)

Ok, I'll move it to the previous line.